src/server/twemcache/data/process.c

```diff
@@ -242,23 +245,25 @@ _process_set(struct response *rsp, struct request *req)
     struct item *it;
     struct bstring key;
 
-    status = _put(&istatus, req);
+    status = _put(&istatus, req, true);
```

Comment:
@seppo0010 I thought about it some more, and can see the appeal of not deleting upon exceptions. Basically it means a command that resulted in error has no side effects. (and client can retry).

The current behavior is inherited from [Twemcache](https://github.com/twitter/twemcache/blob/master/src/mc_ascii.c#L1136 ) (which in turn got it from memcached).